### ISSUE TITLE
IP binding vs token sent addresses conflict on real application

### ISSUE THREAD

**Issue by nokrosis** [wirepair/netcode#13]
I'm using AWS EC2 service to get a test service running.

The example code runs but binding UDPAddr fails using my server public IP

Analyzing the code, when listening UDP addresses, the address is added to a slice in order to listen ports up to `numServers`. 

But in order to correctly listen on the udp address and port correctly, it must use `0.0.0.0` or `127.0.0.1`  as the server IP.

However the client should get the public IP of the server in order to be able to connect to it using the token.

I managed to "cheat" by setting the addresses with the public ip of the server, but binding `netcode.NewServer` to `0.0.0.0`, however, `0.0.0.0` IP gets whitelisted but it's compared against my server public IP so it doesn't match.

I managed to get it to work commenting the whitelist check, but i know it's not the right/secure way.

What's the right way of:

- Binding the address:port successfuly
- Sending the server's public IP in the token
- Whitelisting server's public IP so it matches the connection attempt from the client.

Thanks for your help

TLDR;
Can't bind with public IP.
`0.0.0.0` or `127.0.0.1` won't let clients connect for obvious reasons
Cheating by binding `0.0.0.0` and sending server's public IP won't match whitelist.

**Comment by gafferongames** [comment]
I'm not sure how it works in this version of netcode.io, but in the C version, it always binds to 0.0.0.0 and the server address you specify is the public address that the connect token is whitelist checked against.

If this version of netcode.io doesn't do that, it probably should.

That will fix your issue.

cheers

- Glenn

**Comment by wirepair** [comment]
Fixed by @yome
